contrib/jax_ws/src/python/pants/contrib/jax_ws/tasks/jax_ws_gen.py

```diff
@@ -0,0 +1,91 @@
+# coding=utf-8
+# Copyright 2017 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import logging
+import os
+
+from pants.backend.jvm.targets.java_library import JavaLibrary
+from pants.backend.jvm.tasks.nailgun_task import NailgunTask
+from pants.base.exceptions import TaskError
+from pants.java.distribution.distribution import DistributionLocator
+from pants.task.simple_codegen_task import SimpleCodegenTask
+
+from pants.contrib.jax_ws.targets.jax_ws_library import JaxWsLibrary
+
+logger = logging.getLogger(__name__)
+
+
+class JaxWsGen(SimpleCodegenTask, NailgunTask):
+  """Generates Java files from wsdl files using the JAX-WS compiler."""
+
+  @classmethod
+  def register_options(cls, register):
+    super(JaxWsGen, cls).register_options(register)
+    register('--ws-quiet', type=bool, help='Suppress WsImport output')
+    register('--ws-verbose', type=bool, help='Make WsImport output verbose')
+
+  @classmethod
+  def subsystem_dependencies(cls):
+    return super(JaxWsGen, cls).subsystem_dependencies() + (DistributionLocator,)
+
+  def __init__(self, *args, **kwargs):
+    super(JaxWsGen, self).__init__(*args, **kwargs)
+
+  def synthetic_target_type(self, target):
+    return JavaLibrary
+
+  def is_gentarget(self, target):
+    return isinstance(target, JaxWsLibrary)
+
+  @classmethod
+  def supported_strategy_types(cls):
+    return [cls.IsolatedCodegenStrategy]
+
+  def execute_codegen(self, target, target_workdir):
+    distribution = DistributionLocator.cached(jdk=True)
+    classpath = distribution.find_libs(['tools.jar'])
```

Comment:
This will not work with Java 9 because the tools jar won't exist. Could you add a comment here that this won't work with Java 9? Java 9 dramatically changes the JDK layout.

http://openjdk.java.net/jeps/220 has some of the details. From some googling around, it looks like the way to add these types to the classpath in Java 9 will be to use the `-addmods` option.

I don't think it's blocking as Java9 hasn't been released yet--it's been pushed back to July--but a comment here would be helpful if someone tries to run it with Java 9 and it fails.